pkg/api/v20230401/openshiftcluster_validatestatic.go

```diff
@@ -346,6 +365,45 @@ func (sv openShiftClusterStaticValidator) validateIngressProfile(path string, p
 	return nil
 }
 
+func (sv openShiftClusterStaticValidator) validateResourceTags(path string, t Tags) error {
+	if len(t) > maxTags {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, fmt.Sprintf("The provided set of cluster resource tags is too large; it can contain at most %d tags.", maxTags))
+	}
+
+	var invalidTags []string
+
+	for k, v := range t {
+		if !sv.resourceTagIsValid(k, v) {
+			invalidTags = append(invalidTags, k)
+		}
+	}
+
+	if len(invalidTags) > 0 {
+		errorMsg := "The following cluster resource tags either have an invalid name or an invalid value:\n"
+
+		for _, v := range invalidTags {
+			errorMsg += fmt.Sprintf("- %s\n", v)
+		}
+
+		errorMsg += `
+Tag names must:
+- Start with a letter
+- End with a letter, number, or underscore
+- Contain only letters, numbers, underscores, periods, and hyphens
+- Have length <= 128
+
+Tag values have no character restrictions, but must have length <= 256.`
+
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, errorMsg)
+	}
```

Comment:
You're using two loops when you really only need one I think. Process the error message at the same time as you validate the tag exists.

Reply:
adding on this, your `errorMsg` string starts with a `\n` is that on purpose? 

usually it's better to use a `strings.Builder` than using `+=` for strings because `+=` does a lot more allocations than a builder. 

Here the best would probably be using `strings.Join` to create the string containing all the tags actually.

Reply:
Thanks for the suggestions! I reduced it one loop and used `strings.Join()` instead of concatenation.